### PR TITLE
Issue 1701 1d log10 axes

### PR DESCRIPTION
**Pull request by doutriaux1**
@aashish24 @sankhesh please review, we have users that need this in 2.4

### CODE REVIEW

**Comment by doutriaux1** [comment]
goes with:  https://github.com/UV-CDAT/uvcdat-testdata/pull/86

**Comment by aashish24** [comment]
@doutriaux1 do we need this?

File: Packages/vcs/Lib/vcsvtk/pipeline1d.py [hunk 1]
```diff
@@ -21,6 +33,65 @@ class Pipeline1D(Pipeline):
     def __init__(self, gm, context_):
         super(Pipeline1D, self).__init__(gm, context_)
 
+    def convertAxis(self, axis, data, method, revert_method):
+        new = method(data[:])
```
Comment (sankhesh) [inline Packages/vcs/Lib/vcsvtk/pipeline1d.py]: @doutriaux1 `new` is a very bad variable name

File: Packages/vcs/Lib/vcsvtk/pipeline1d.py [hunk 2]
```diff
@@ -39,6 +110,24 @@ def plot(self, data1, data2, tmpl, grid, transform):
         if self._gm.smooth is not None:
             Y = smooth(Y, self._gm.smooth)
 
+        # Here we try to apply the axis convert options
+        for ax, conv, vals in [['x', self._gm.xaxisconvert, X], ['y', self._gm.yaxisconvert, Y]]:
+            if conv == "log10":
+                new = self.convertAxis(ax, vals, numpy.ma.log10, power10)
```
Comment (sankhesh) [inline Packages/vcs/Lib/vcsvtk/pipeline1d.py]: Same comment about `new` here.

File: Packages/vcs/Lib/vcsvtk/pipeline1d.py [hunk 3]
```diff
@@ -39,6 +110,24 @@ def plot(self, data1, data2, tmpl, grid, transform):
...
+            elif conv == "ln":
+                new = self.convertAxis(ax, vals, numpy.ma.log, numpy.ma.exp)
```
Comment (sankhesh) [inline Packages/vcs/Lib/vcsvtk/pipeline1d.py]: Why have a `method` and a `revert_method`?

Can't the `revert_method` be inferred from `method`?

**Comment by sankhesh** [comment]
@doutriaux1 Test failures

**Comment by doutriaux1** [comment]
No we do not need this in this release it is preliminary work, being tested by users here, I mostly created the PR to see if it broke anything, no need to review.

**Comment by aashish24** [comment]
@doutriaux1 I am closing this now unless you decided to work on it again (and at that time you can open it again).

**Comment by doutriaux1** [comment]
Ok, funny you’re doing it now, the user actually came into my office about it today ;) Will probably work on it next week, it’s needed for 2.6

**Pull request closed without merging**